enhancements/kube-apiserver/event-ttl.md

```diff
@@ -0,0 +1,275 @@
+
+There were no long-term increases in CPU/memory usage found after configuring a 5m TTL vs. the existing default. 
+
+### Risks and Mitigations
+
+**Risk**: Customers might set the value to 0, which means that the events will never be deleted
+**Mitigation**: The API validation and operator ensures the values are within a reasonable range and never zero (5-180 minutes).
+
+### Drawbacks
+
+- Adds complexity to the configuration API
+- Additional validation and error handling required
+
+## Alternatives (Not Implemented)
+
+1. **Hardcoded Values**: Keep the current fixed value of 3 hours
+   - **Rejected**: Does not meet customer requirements for configurability
+
+2. **Environment Variable**: Use environment variables instead of API configuration
+   - **Rejected**: Less user-friendly and harder to manage
+
+3. **Separate CRD**: Create a separate CRD for event configuration
+   - **Rejected**: Overkill for a single setting, better to include in existing APIServer resource
+
+## Test Plan
+
+The test plan will include:
+
+1. **Unit Tests**: Test the API validation and parsing logic
+2. **E2E Tests**: Test that the event TTL is properly configured on all apiserver after applying the setting
+3. **Performance Tests**: Test the impact of different TTL values on etcd performance
+
+## Tech Preview
+
+The EventTTL feature is controlled by the `EventTTL` feature gate, which is enabled by default in both DevPreview and TechPreview feature sets. This allows the feature to be available for testing and evaluation without requiring additional configuration.
+
+The EventTTL feature gate is implemented in [openshift/api PR #2525](https://github.com/openshift/api/pull/2525) and will be removed when the feature graduates to GA, as the functionality will become a standard part of the platform.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+- API is implemented and validated
+- Basic functionality works end-to-end
+- Documentation is available
+- Sufficient test coverage
+- EventTTL feature gate is enabled in DevPreview and TechPreview feature sets
+
+### Tech Preview -> GA
+
+- More comprehensive testing (upgrade, downgrade, scale)
+- Performance testing with various TTL values
+- User feedback incorporated
+- Documentation updated in openshift-docs
+- EventTTL feature gate is removed as the feature becomes GA
+
+### Removing a deprecated feature
+
+This enhancement does not remove any existing features. It only adds new configuration options while maintaining backward compatibility with the existing default behavior.
+
+## Upgrade / Downgrade Strategy
+
+### Upgrade Strategy
+
+- Existing clusters will continue to use the default 3-hour (180-minute) TTL
+- No changes required for existing clusters
+- New configuration option is available immediately
+
+### Downgrade Strategy
+
+- Configuration will be ignored by older versions
```

Comment:
Will downgraded clusters with the older CRDs still be able to deserialize the stored-in-etcd resources?  I'm not sure how that handles unrecognized properties in etcd.  But even if this breaks, we don't support downgrading from 4.y to 4.(y-1), so it would only be a brief hiccup for rollbacks within 4.dev Engineering Candidates and such.